src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -602,6 +663,14 @@ ucs_status_t uct_ib_mlx5_devx_uar_init(uct_ib_mlx5_devx_uar_t *uar,
 void uct_ib_mlx5_devx_uar_cleanup(uct_ib_mlx5_devx_uar_t *uar)
 {
 #if HAVE_DEVX
+    if (uar == NULL) {
```

Comment:
it shouldn't be
uct_ib_mlx5_devx_uar_cleanup() problem, the caller shouldn't call it with NULL

Reply:
So do we need this check in the caller?

Reply:
we don't need to check for NULL in this function, a NULL value would be bug in the code and we do not need to have protection against it

Reply:
@yosefe let's keep at least the second one (the internal `uar` check)..

Reply:
removed the first null check

Reply:
removed the second null check as well @yosefe @Artemy-Mellanox